packages/web/src/pages/sign-up-page/pages/CreateEmailPage/CreateEmailPage.tsx

```diff
@@ -0,0 +1,96 @@
+import { useCallback, useContext } from 'react'
+
+import { AudiusQueryContext, signUpFetch } from '@audius/common'
+import { Form, Formik, FormikHelpers } from 'formik'
+import { useDispatch, useSelector } from 'react-redux'
+import { z } from 'zod'
+import { toFormikValidationSchema } from 'zod-formik-adapter'
+
+import { setValueField } from 'common/store/pages/signon/actions'
+import { getEmailField } from 'common/store/pages/signon/selectors'
+import { useMedia } from 'hooks/useMedia'
+import { useNavigateToPage } from 'hooks/useNavigateToPage'
+import { EMAIL_REGEX } from 'utils/email'
+import { SIGN_IN_PAGE, SIGN_UP_PASSWORD_PAGE } from 'utils/route'
+
+import styles from './CreateEmailPage.module.css'
+import { CreateEmailPageDesktop } from './CreateEmailPageDesktop'
+import { CreateEmailPageMobile } from './CreateEmailPageMobile'
+import { messages } from './messages'
+
+type SignUpEmailValues = {
+  email: string
+}
+
+const FormSchema = toFormikValidationSchema(
+  z.object({
+    email: z
+      .string({ required_error: messages.invalidEmail })
+      .regex(EMAIL_REGEX, { message: messages.invalidEmail })
+  })
+)
+
+/**
+ * Component responsible for form logic and controlling whether to show mobile/desktop.
+ * UI render logic is split out into separate components for mobile/desktop
+ */
+export const CreateEmailPage = () => {
```

Comment:
i like the page change name... its a bit weird with "create-email"  since we arent creating.... create-from-email? anyway i dont have a great alternative

Reply:
Yeah, not sure if there's a better wording, but thats also what's in the route name as well so it matches for now at least